File: examples/python/scan_example.py
"""
Surmado Scan - SEO Audit Example
Run an SEO audit on any website.
Docs: https://help.surmado.com/docs/api-reference/
"""
import os
import time
import requests

API_KEY = os.environ.get("SURMADO_API_KEY", "YOUR_API_KEY")
BASE_URL = "https://api.surmado.com/v1"


def create_scan_report(url: str, brand_name: str, email: str, tier: str = "basic"):
    """
    Create a Scan (SEO audit) report.
    
    Args:
        url: Website URL to audit
        brand_name: Your brand name
        email: Email for notifications
        tier: "basic" (1 credit, $25) or "premium" (2 credits, $50)
    
    Returns:
        Report data with report_id
    """
    response = requests.post(
        f"{BASE_URL}/reports/scan",
        headers={
            "X-API-Key": API_KEY,
            "Content-Type": "application/json",
        },
        json={
            "url": url,
            "brand_name": brand_name,
            "email": email,
            "tier": tier,
        },
    )
    response.raise_for_status()
    return response.json()


def get_report(report_id: str):
    """Get report status and results."""
    response = requests.get(
        f"{BASE_URL}/reports/{report_id}",
        headers={"X-API-Key": API_KEY},
    )
    response.raise_for_status()
    return response.json()


def wait_for_completion(report_id: str, timeout_minutes: int = 20):
    """
    Poll until report completes or fails.
    
    Args:
        report_id: The report ID to poll
        timeout_minutes: Maximum time to wait (default 20 min)
    
    Returns:
        Completed report data
    """
    start = time.time()
    timeout_seconds = timeout_minutes * 60
    
    while time.time() - start < timeout_seconds:
        report = get_report(report_id)
        status = report.get("status")
        
        if status == "completed":
            print(f"Report completed in {int(time.time() - start)} seconds")
            return report
        elif status == "failed":
            raise Exception(f"Report failed: {report.get('error')}")
        
        print(f"Status: {status}. Waiting...")
        time.sleep(30)  # Poll every 30 seconds
    
    raise TimeoutError(f"Report did not complete within {timeout_minutes} minutes")


if __name__ == "__main__":
    # Create a Scan report
    print("Creating Scan report...")
    result = create_scan_report(
        url="https://example.com",
        brand_name="Example Brand",
        email="you@example.com",
        tier="basic",
    )
    
    report_id = result["report_id"]
    print(f"Report created: {report_id}")
    print(f"Credits used: {result.get('credits_used')}")
    
    # Wait for completion
    print("\nWaiting for report to complete...")
    completed = wait_for_completion(report_id)
    
    # Download results
    if completed.get("download_url"):
        print(f"\nPDF: {completed['download_url']}")
    if completed.get("intelligence_download_url"):
        print(f"JSON: {completed['intelligence_download_url']}")

File: examples/javascript/scan_example.js
/**
 * Surmado Scan - SEO Audit Example
 * Run an SEO audit on any website.
 * Docs: https://help.surmado.com/docs/api-reference/
 */

const API_KEY = process.env.SURMADO_API_KEY || 'YOUR_API_KEY';
const BASE_URL = 'https://api.surmado.com/v1';

/**
 * Create a Scan (SEO audit) report.
 * @param {string} url - Website URL to audit
 * @param {string} brandName - Your brand name
 * @param {string} email - Email for notifications
 * @param {string} tier - "basic" (1 credit, $25) or "premium" (2 credits, $50)
 * @returns {Promise<Object>} Report data with report_id
 */
async function createScanReport(url, brandName, email, tier = 'basic') {
  const response = await fetch(`${BASE_URL}/reports/scan`, {
    method: 'POST',
    headers: {
      'X-API-Key': API_KEY,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      url,
      brand_name: brandName,
      email,
      tier,
    }),
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API error: ${JSON.stringify(error)}`);
  }

  return response.json();
}

/**
 * Get report status and results.
 * @param {string} reportId - The report ID to check
 * @returns {Promise<Object>} Report data
 */
async function getReport(reportId) {
  const response = await fetch(`${BASE_URL}/reports/${reportId}`, {
    headers: { 'X-API-Key': API_KEY },
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API error: ${JSON.stringify(error)}`);
  }

  return response.json();
}

/**
 * Poll until report completes or fails.
 * @param {string} reportId - The report ID to poll
 * @param {number} timeoutMinutes - Maximum time to wait (default 20 min)
 * @returns {Promise<Object>} Completed report data
 */
async function waitForCompletion(reportId, timeoutMinutes = 20) {
  const start = Date.now();
  const timeoutMs = timeoutMinutes * 60 * 1000;

  while (Date.now() - start < timeoutMs) {
    const report = await getReport(reportId);
    const status = report.status;

    if (status === 'completed') {
      console.log(`Report completed in ${Math.round((Date.now() - start) / 1000)} seconds`);
      return report;
    } else if (status === 'failed') {
      throw new Error(`Report failed: ${report.error}`);
    }

    console.log(`Status: ${status}. Waiting...`);
    await new Promise(resolve => setTimeout(resolve, 30000)); // Poll every 30 seconds
  }

  throw new Error(`Report did not complete within ${timeoutMinutes} minutes`);
}

// Main example
async function main() {
  try {
    // Create a Scan report
    console.log('Creating Scan report...');
    const result = await createScanReport(
      'https://example.com',
      'Example Brand',
      'you@example.com',
      'basic'
    );

    const reportId = result.report_id;
    console.log(`Report created: ${reportId}`);
    console.log(`Credits used: ${result.credits_used}`);

    // Wait for completion
    console.log('\nWaiting for report to complete...');
    const completed = await waitForCompletion(reportId);

    // Download results
    if (completed.download_url) {
      console.log(`\nPDF: ${completed.download_url}`);
    }
    if (completed.intelligence_download_url) {
      console.log(`JSON: ${completed.intelligence_download_url}`);
    }
  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
}

main();


